source/slang/slang-ir-util.cpp

```diff
@@ -1557,10 +1557,24 @@ void hoistInstOutOfASMBlocks(IRBlock* block)
 IRType* getSPIRVSampledElementType(IRInst* sampledType)
 {
     auto sampledElementType = getVectorElementType((IRType*)sampledType);
-    if (sampledElementType->getOp() == kIROp_HalfType)
+    
+    IRBuilder builder(sampledType);
+    switch (sampledElementType->getOp())
     {
-        IRBuilder builder(sampledType);
+    case kIROp_HalfType:
         sampledElementType = builder.getBasicType(BaseType::Float);
+        break;
+    case kIROp_UInt16Type:
```

Comment:
for half/uint16/int16, should we be emitting a `RelaxedPrecision` decoration (the value is still 32bit so we would just be adding the decoration)?

Reply:
That decoration will need to be added to the sampling instructions, and needs to be propagated through computations. I am not sure how helpful it will be but seems like non trivial amount of effort so I would skip it unless someone tells us that is an issue.